src/components/application_manager/test/commands/mobile/add_command_request_test.cc

```diff
@@ -0,0 +1,299 @@
+  ON_CALL(app_mngr_, application(_)).WillByDefault(Return(mock_app));
+
+  EXPECT_CALL(*mock_app, FindCommand(_));
+  EXPECT_CALL(*mock_app, AddCommand(_, _));
+
+  MockMessageHelper* mock_message_helper =
+      MockMessageHelper::message_helper_mock();
+  ON_CALL(*mock_message_helper, VerifyImage(_, _, _))
+      .WillByDefault(Return(mobile_apis::Result::SUCCESS));
+
+  application_manager::CommandsMap commands_map;
+  ON_CALL(*mock_app, commands_map())
+      .WillByDefault(Return(
+          DataAccessor<application_manager::CommandsMap>(commands_map, lock_)));
+
+  req->Run();
+}
+
+TEST_F(AddCommandRequestTest, OnEvent_UI_SUCCESS) {
+  MessageSharedPtr msg_ui = create_fullParams_ui_so();
+  using namespace application_manager::commands;
+  utils::SharedPtr<AddCommandRequest> req_ui =
+      CreateCommand<AddCommandRequest>(msg_ui);
+
+  MockAppPtr mock_app = CreateMockApp();
+  ON_CALL(app_mngr_, application(_)).WillByDefault(Return(mock_app));
+  ON_CALL(*mock_app, app_id()).WillByDefault(Return(1));
+
+  MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+  (*msg)[strings::params][hmi_response::code] =
+      hmi_apis::Common_Result::SUCCESS;
+  (*msg)[strings::msg_params][strings::cmd_id] = 1;
+
+  am::event_engine::Event event(hmi_apis::FunctionID::UI_AddCommand);
+  event.set_smart_object(*msg);
+
+  smart_objects::SmartObject* ptr = NULL;
+  ON_CALL(*mock_app, FindCommand(_)).WillByDefault(Return(ptr));
+
+  MockMessageHelper* mock_message_helper =
+      MockMessageHelper::message_helper_mock();
+  ON_CALL(*mock_message_helper, HMIToMobileResult(_))
+      .WillByDefault(Return(mobile_apis::Result::SUCCESS));
+
+  EXPECT_CALL(*mock_app, AddCommand(_, _));
+  EXPECT_CALL(*mock_app, UpdateHash());
+
+  am::CommandsMap commands_map;
+  ON_CALL(*mock_app, commands_map())
+      .WillByDefault(
+          Return(DataAccessor<am::CommandsMap>(commands_map, lock_)));
+
+  req_ui->Run();
+  req_ui->on_event(event);
+}
+
+TEST_F(AddCommandRequestTest, OnEvent_VR_SUCCESS) {
+  MessageSharedPtr msg_vr = create_fullParams_vr_so();
+  using namespace application_manager::commands;
+  utils::SharedPtr<AddCommandRequest> req_vr =
+      CreateCommand<AddCommandRequest>(msg_vr);
+
+  MockAppPtr mock_app = CreateMockApp();
+  ON_CALL(app_mngr_, application(_)).WillByDefault(Return(mock_app));
+  ON_CALL(*mock_app, app_id()).WillByDefault(Return(1));
+
+  MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+  (*msg)[strings::params][hmi_response::code] =
+      hmi_apis::Common_Result::SUCCESS;
+  (*msg)[strings::msg_params][strings::cmd_id] = 1;
+
+  am::event_engine::Event event(hmi_apis::FunctionID::VR_AddCommand);
+  event.set_smart_object(*msg);
+
+  smart_objects::SmartObject* ptr = NULL;
+  ON_CALL(*mock_app, FindCommand(_)).WillByDefault(Return(ptr));
+
+  MockMessageHelper* mock_message_helper =
+      MockMessageHelper::message_helper_mock();
+  ON_CALL(*mock_message_helper, HMIToMobileResult(_))
```

Comment:
@wolfylambova22, please, replace `ON_CALL`s for `MockMessageHelper` by `EXPECT_CALL`s. Other ways it can cost an unpredictable behavior.